Mathlib/RingTheory/AdicCompletion/Functoriality.lean

```diff
@@ -0,0 +1,352 @@
+/-
+Copyright (c) 2024 Judith Ludwig, Christian Merten. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Judith Ludwig, Christian Merten
+-/
+import Mathlib.RingTheory.AdicCompletion.Basic
+import Mathlib.RingTheory.AdicCompletion.Algebra
+import Mathlib.Algebra.DirectSum.Basic
+
+/-!
+# Functoriality of adic completions
+
+In this file we establish functorial properties of the adic completion.
+
+## Main definitions
+
+- `LinearMap.adicCauchy I f`: the by `f` induced linear map on `I`-adic cauchy sequences
+- `LinearMap.adicCompletion I f`: the by `f` induced linear map on `I`-adic completions
+
+## Main results
+
+- `sumEquivOfFintype`: adic completion commutes with finite sums
+- `piEquivOfFintype`: adic completion commutes with finite products
+
+-/
+
+variable {R : Type*} [CommRing R] (I : Ideal R)
+variable {M : Type*} [AddCommGroup M] [Module R M]
+variable {N : Type*} [AddCommGroup N] [Module R N]
+variable {P : Type*} [AddCommGroup P] [Module R P]
+variable {T : Type*} [AddCommGroup T] [Module (AdicCompletion I R) T]
+
+namespace LinearMap
+
+/-- `R`-linear version of `reduceModIdeal`. -/
+private def reduceModIdealAux (f : M →ₗ[R] N) :
+    M ⧸ (I • ⊤ : Submodule R M) →ₗ[R] N ⧸ (I • ⊤ : Submodule R N) :=
+  Submodule.mapQ (I • ⊤ : Submodule R M) (I • ⊤ : Submodule R N) f
+    (fun x hx ↦ by
+      refine Submodule.smul_induction_on hx (fun r hr x _ ↦ ?_) (fun x y hx hy ↦ ?_)
+      · simp [Submodule.smul_mem_smul hr Submodule.mem_top]
+      · simp [Submodule.add_mem _ hx hy])
+
+@[simp]
```

Comment:
```suggestion
@[local simp]
```

Reply:
Did you really use this somewhere? I doubt so.

Reply:
No, they only appear in `simp only` calls. I changed it to `local simp` as suggested, should I remove it completely instead?